src/gui_common/dialogs/CustomDialog.cs

```diff
@@ -395,6 +442,38 @@ protected override void ApplyRectSettings()
         }
     }
 
+    protected void ConnectToWindowReorderingNodes()
+    {
+        var windowReorderingAncestorsIEnumerable = AddWindowReorderingSupportToSiblings.GetWindowReorderingAncestors(
+            this, AutomaticWindowReorderingDepth, WindowReorderingPaths);
+
+        foreach (var windowReordering in windowReorderingAncestorsIEnumerable)
+        {
+            windowReordering.ReorderingNode.ConnectWindow(
+                this, windowReordering.Sibling, AllowWindowReorderingRecursion);
+            windowReorderingNodes.Add(windowReordering.ReorderingNode);
+        }
+    }
+
+    protected void DisconnectFromWindowReorderingNodes()
+    {
+        foreach (var node in windowReorderingNodes)
+            node.DisconnectWindow(this, AllowWindowReorderingRecursion);
```

Comment:
Maybe `AllowWindowReorderingRecursion` should have a warning that it may not be changed while this is attached to the node tree? As that will cause a bug if someone sets that to false after the connections are made and then removes this from the scene tree.

Reply:
What about writing a getter and setter with a logic to prevent it from breaking?

Reply:
That could be doable as well, but the logic to reapply the different values would be pretty complicated as unregistering and re-registering with the right recursive type would be needed.

An easier fix I also came up with was storing the original value of window recursion when registering, then that variable could be used here independently of whether AllowWindowReorderingRecursion was changed or not. If this approach is taken a comment should be added to AllowWindowReorderingRecursion to say that it doesn't apply immediately if the node is currently in use.

Reply:
There are already methods that do the registering and unregistering so it's also simple.

It's more about how it should behave and I am not sure what is the right behaviour. Is there ever a reason to change it in the code? We can make it readonly if this is the case, otherwise your approach is probably better because in the current tree it's usually not expected for it to change the behaviour, I think.

Reply:
Well you can't have the Godot set stuff be readonly. And anyway the paths *will* change if nothing else the modal system will yank windows out from the scene tree and place them in completely different places. Though maybe that actually uncovers a bug here: the relative paths (and absolute if they are resolved) will definitely break when the window is moved to be a child of the modal popup layer when it becomes a modal.

Reply:
It worked when I set them to readonly. Godot allows to change them in the editor.
The list of paths should contain only relative paths if I am not mistaken? (I don't have any check for if they are relative or absolute) Also they shouldn't break. It skips connecting to non-existent nodes.

There isn't a way to properly change the paths though. If this is needed then I can also apply the changes after re-entering the tree and not immediately.

Reply:
If you put a `readonly` modifier on it, that's not good enough. This is C# so while the reference can't be updated to point to a different object, the list object is still modifiable by any piece of code.

>  (I don't have any check for if they are relative or absolute) Also they shouldn't break. It skips connecting to non-existent nodes.

I guess I'd need to read the code to figure out. But then why was there a need to cache the ancestor nodes for checking? If that kind of relative node check could just be done from the beginning?

Also if anyone ever uses a code approach to setup a window, they might have an absolute path set as those are easier to work with when modifying nodes in the tree as keeping relative paths up to date is much more difficult when manually manipulating things through code.

> There isn't a way to properly change the paths though. If this is needed then I can also apply the changes after re-entering the tree and not immediately.

The paths is a public property so changing its contents should work, similarly to why I wrote the FocusGrabber to redo its paths.

Reply:
I meant readonly for the AllowWindowReorderingRecursion, but I think that applying it later is a better approach.

> I guess I'd need to read the code to figure out. But then why was there a need to cache the ancestor nodes for checking? If that kind of relative node check could just be done from the beginning?

If you mean the part with the dictionary in the GetWindowReorderingAncestors method then it was just a bad code. The complications come with finding the right sibling.

> Also if anyone ever uses a code approach to setup a window, they might have an absolute path set as those are easier to work with when modifying nodes in the tree as keeping relative paths up to date is much more difficult when manually manipulating things through code.

Well then I will have count with that too.

>The paths is a public property so changing its contents should work, similarly to why I wrote the FocusGrabber to redo its paths.

Actually there is no need for additional logic for that. The paths are not used when exiting a tree so they can be changed directly and the change will take effect after re-entering the tree.

Reply:
Guess I need to read your latest changes before I can offer useful insight. At the latest I'll have time in a couple of days to do that.

Reply:
After looking at the code again I came to the conclusion that there is nothing that shouldn't allow absolute paths.
So I will just change AllowWindowReorderingRecursion to update after re-entering.